forms/forms100.py

```diff
@@ -846,6 +849,50 @@ def form_04(request_data):
     objs.extend(content_title)
     objs.append(Spacer(1, 5 * mm))
 
+    research_need = DispensaryPlan.objects.filter(diagnos=diagnos).order_by('research__title', 'speciality__title')
+    researches_list = []
+    specialities_list = []
+    visits_result = []
+    visits_research = VisitPurpose.objects.filter(title__icontains="диспансерн")
+
+    year = '2020'
+    for i in research_need:
+        if i.speciality:
+            results = research_last_result_every_month(Researches.objects.filter(speciality=i.speciality), ind_card, year, visits_research)
+            dates_result = " "
+            dates_plan = " "
+            plans = DispensaryRegPlans.objects.filter(card=ind_card, research=None, speciality=i.speciality, date__year='2020')
+            for p in plans:
+                date = strdate(p.date)
+                dates_plan = f"{dates_plan} {normalize_date(date)[0:5]};"
```

Comment:
Зачем делать слайс?
И приводить дату к строке?
Почему не сделать тут так?
```suggestion
                dates_plan = f"{dates_plan} {p.date.year};"
```

Reply:
здесь берется не год. Т.к . на каждый год карта своя то нужно число месяц в формате 01.03. Если 
p.date.month p.date.day - то отдается 2 и 9 без нулей - придется добавлять нули спереди.  normalize_date - быстрее это сделать

Reply:
Тут совсем не понятно, что происходит
Лучше сделать понятнее, что такое этот [0:5] – можно понять, только всё досконально изучив
Почему не сделать сразу тогда нужную строку, не "нормализовать дату"
У нас есть утилита – strfdatetime можно передать шаблон, какую дату надо получить, например
```suggestion
                dates_plan = f"{dates_plan} {strfdatetime(p.date, '%d.%m')};"
```